### ISSUE TITLE
az aks get-credentials [...] fails with "'ManagedCluster' object has no attribute 'properties'"

### ISSUE THREAD

**Issue by frankbuckley** [Azure/AKS#113]
Might be since Azure CLI 2.0.24:

	PS C:\> az aks get-credentials -g [group] -n [name]
	'ManagedCluster' object has no attribute 'properties'
	Traceback (most recent call last):
	  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\site-packages\knack\cli.py", line 193, in invoke cmd_result = self.invocation.execute(args)
	  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\site-packages\azure\cli\core\commands\__init__.py", line 319, in execute six.reraise(*sys.exc_info())
	  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\site-packages\six.py", line 693, in reraise raise value
	  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\site-packages\azure\cli\core\commands\__init__.py", line 292, in execute result = cmd(params)
	  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\site-packages\azure\cli\core\commands\__init__.py", line 169, in __call__return super(AzCliCommand, self).__call__(*args, **kwargs)
	  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\site-packages\knack\commands.py", line 109, in __call__return self.handler(*args, **kwargs)
	  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\site-packages\azure\cli\core\__init__.py", line 328, in default_command_handler result = op(**command_args)
	  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\site-packages\azure\cli\command_modules\acs\custom.py", line 1329, in aks_get_credentialsaccess_profiles = mc.properties.access_profiles
	AttributeError: 'ManagedCluster' object has no attribute 'properties'

Seen with two clusters in different accounts. Not tried rolling back Azure CLI.

**Comment by frankbuckley** [comment]
My Mac still has CLI Azure CLI 2.0.23 and `az aks get-credentials` works OK there.

(Previous failures were on Windows 10 with Azure CLI 2.0.24.)

**Comment by mboersma** [comment]
The 2.0.24 CLI release includes a large refactoring that apparently lost some code changes AKS made in 2.0.23. For now, the only workaround is to stick with 2.0.23.

The core issue is here: https://github.com/Azure/azure-cli/issues/5223

**Comment by vpopescu** [comment]
"the core issue" is set to Closed? Meaning?

**Comment by mboersma** [comment]
That issue was closed when the fix in Azure/azure-cli#5226 was merged, but it isn't available in an installable release of `az` yet. According to [the milestone page](https://github.com/Azure/azure-cli/milestones) it should be available on January 17, in version 2.0.25 presumably.

**Comment by karolz-ms** [comment]
This is a real nuisance on MacOS because with Homebrew there is no good way to go back to an older version if you do not have it already installed.

**Comment by KarthickSelvam** [comment]
What to do till then?

**Comment by sdhamilton6** [comment]
@karolz-ms you can install the 2.0.23 pull request directly from brew:
brew install https://raw.githubusercontent.com/derekbekoe/homebrew-core/000e0c71acaafd20f9305e84f19991095c88ee67/Formula/azure-cli.rb
After uninstalling 2.0.24 off course :)

**Comment by karolz-ms** [comment]
Good to know, thanks!!

I did some more poking around yesterday and found this too https://stackoverflow.com/questions/39187812/homebrew-how-to-install-older-versions A bit gnarly, but worked for me

**Comment by mboersma** [comment]
And if you're a developer and have Docker installed, you can [easily run specific `az` versions](https://github.com/Azure/azure-cli#docker):
```shell
$ docker run -v ${HOME}:/root -it azuresdk/azure-cli-python:2.0.23
```

**Comment by ryanmentzer** [comment]
For Windows users, the 2.0.23 MSI can be found here: [https://azurecliprod.blob.core.windows.net/msi/azure-cli-2.0.23.msi](https://azurecliprod.blob.core.windows.net/msi/azure-cli-2.0.23.msi).

**Comment by andrewb-ms** [comment]
This is so broken!

**Comment by ilyabelkin** [comment]
Here are the steps to install 2.0.23 with Homebrew on Mac:

- Uninstall the broken version (2.0.24; the problem should be fixed in 2.0.25 that is scheduled to be released on Jan 17)
- Using homebrew, install version 2.0.23
```
brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/d8a64440c3944f1dfde4f1bb228d7218868c636f/Formula/azure-cli.rb
```
If you get the python3 symlink error, run the following commands (this assumes you don't have the Frameworks folder created yet):
```
sudo mkdir /usr/local/Frameworks
sudo chown $(whoami):admin /usr/local/Frameworks
sudo brew link python3
```

**Comment by frankbuckley** [comment]
Fixed in azure-cli 2.0.25
